assembly/src/ast/nodes/format.rs

```diff
@@ -0,0 +1,151 @@
+use super::{AstFormatterContext, FormattableCodeBody, Instruction, Node};
+use core::fmt;
+
+// FORMATTING OF NODES
+// ================================================================================================
+pub struct FormattableNode<'a> {
+    node: &'a Node,
+    context: &'a AstFormatterContext<'a>,
+}
+
+impl<'a> FormattableNode<'a> {
+    pub fn new(node: &'a Node, context: &'a AstFormatterContext<'a>) -> Self {
+        Self { node, context }
+    }
+}
+
+impl fmt::Display for FormattableNode<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.node {
+            Node::Instruction(i) => {
+                write!(f, "{}", FormattableInstruction::new(i, self.context))
+            }
+            Node::IfElse {
+                true_case,
+                false_case,
+            } => {
+                self.context.indent(f)?;
+                writeln!(f, "if.true")?;
+                write!(
+                    f,
+                    "{}",
+                    FormattableCodeBody::new(true_case, &self.context.inner_scope_context())
+                )?;
+                if false_case.nodes().is_empty() {
```

Comment:
Should this be `if !false_case.nodes().is_empty()`?

Reply:
One would think that ~25 years of practice would be enough to learn Boolean logic, but apparently not. :-(

Fixed. Good catch.